File: benchmark/c/cpw/dynamic_programming_set_36_cut_a_rope_to_maximize_product_1/dynamic_programming_set_36_cut_a_rope_to_maximize_product_1.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int dynamic_programming_set_36_cut_a_rope_to_maximize_product_1 ( int n ) ;
int dynamic_programming_set_36_cut_a_rope_to_maximize_product_1 ( int n ) {
  if ( n == 2 || n == 3 ) return ( n - 1 );
  int res = 1;
  while ( n > 4 ) {
    n -= 3;
    res *= 3;
  }
  return ( n * res );
}

File: translation_result/else_bms/fail/c/cpw/dynamic_programming_set_36_cut_a_rope_to_maximize_product_1/gemini/Restart/dynamic_programming_set_36_cut_a_rope_to_maximize_product_1.rs

fn dynamic_programming_set_36_cut_a_rope_to_maximize_product_1(mut n: i32) -> i32 {
    if n == 2 || n == 3 {
        return n - 1;
    }
    let mut res = 1;
    while n > 4 {
        n -= 3;
        res *= 3;
    }
    n * res
}
